workspaces/arborist/lib/arborist/build-ideal-tree.js

```diff
@@ -1475,6 +1485,32 @@ This is a one-time fix-up, please be patient...
     timeEnd()
   }
 
+  #applyRootOverridesToWorkspaces (tree) {
+    const rootOverrides = tree.root.package.overrides || {}
+
+    for (const node of tree.root.inventory.values()) {
+      if (!node.isWorkspace) {
+        continue
+      }
+
+      for (const depName of Object.keys(rootOverrides)) {
+        const edge = node.edgesOut.get(depName)
+        const rootNode = tree.root.children.get(depName)
+
+        // safely skip if either edge or rootNode doesn't exist yet
+        if (!edge || !rootNode) {
+          continue
+        }
+
+        const resolvedRootVersion = rootNode.package.version
+        if (!semver.satisfies(resolvedRootVersion, edge.spec)) {
+          edge.detach()
+          node.children.delete(depName)
+        }
```

Comment:
If we remove the edge/children from the tree if it's not satisfied by an override, then we're assuming that the package is elsewhere in the tree already. I can imagine a case where that isn't true.

```
- package.json
  - override package-x@^1.0.0
  - dependences { doesn't include package-x }
- workspace/
  - package.json
    - dependencies {does include package-x@^2.0.0 }
```
In this case, the tree would not include the required dependency once this function removes it from the tree.

Is that possible?